### ISSUE TITLE
Elemental ISO Generated without callback URL [BUG]

### ISSUE THREAD

**Issue by rdconnelly** [rancher/rancher#42591]
**Rancher Server Setup**
- Rancher version: v2.7.3
- Installation option (Docker install/Helm Chart): Helm v2.16.8-rancher2
- Proxy/Cert Details: n/a

**Information about the Cluster**
- Kubernetes version:
Client Version: v1.27.2
Kustomize Version: v5.0.1
Server Version: v1.27.4-eks-2d98532

- Cluster Type (Local/Downstream):
   - If downstream, what type of cluster? (Custom/Imported or specify provider for Hosted/Infrastructure Provider):

**User Information**
- What is the role of the user logged in? Admin
  - If custom, define the set of permissions: n/a

**Describe the bug**

The elemental ISO does not contain a callback URL to Rancher.  Content of the  LIVECD_CLOUD_CONFIG.yaml is listed below. This file is at the root directory of the ISO. 

```
elemental:
    registration:
        ca-cert: |
            -----BEGIN CERTIFICATE-----
            MIIBvTCCAWOgAwIBAgIBADAKBggqhkjOPQQDAjBGMRwwGgYDVQQKExNkeW5hbWlj
            bGlzdGVuZXItb3JnMSYwJAYDVQQDDB1keW5hbWljbGlzdGVuZXItY2FAMTY4NDAx
            OTQ4MTAeFw0yMzA1MTMyMzExMjFaFw0zMzA1MTAyMzExMjFaMEYxHDAaBgNVBAoT
            E2R5bmFtaWNsaXN0ZW5lci1vcmcxJjAkBgNVBAMMHWR5bmFtaWNsaXN0ZW5lci1j
            YUAxNjg0MDE5NDgxMFkwEwYHKoZIzj0CAQYIKoZIzj0DAQcDQgAEWg7T8g0xdR6c
            PqBkYe7XIpvic+lf91ZQ4UHSjOx8TGG1UoE2hfM1sXfcz5d4Df3ltnkFxzGzXKfR
            taLaSCuyUqNCMEAwDgYDVR0PAQH/BAQDAgKkMA8GA1UdEwEB/wQFMAMBAf8wHQYD
            VR0OBBYEFP+dYnUy607TQgS6GEOdvwQURyf9MAoGCCqGSM49BAMCA0gAMEUCIDIZ
            4xu9Fy0ZDl4IIwg+2JWLz+nZgpQOKjf8oUyR2TThAiEAuaIlv6B7wbPg1RXO7hol
            bAWW4YDF/W5frbUwC20/TDk=
            -----END CERTIFICATE-----
        emulate-tpm: true
        emulated-tpm-seed: -1
        no-smbios: true
        auth: tpm
```

When a VM is booted with this ISO, the elemental-register.service complains the 'Registration URL is empty'

**To Reproduce**
Create a Kubernetes [SeedImage](https://elemental.docs.rancher.com/seedimage-reference) resource and download the elemental ISO when  it becomes available. Boot a VM using this iso and check the logs of  elemental-register.service.

We have created ~100 clusters using the exact same process and have only come across this issue once.

**Result**

**Expected Result**
The elemental ISO should have the callback URL.  Within the ISO, the  LIVECD_CLOUD_CONFIG.yaml file should have the callback URL. For example:

```
elemental:
    registration:
        url: https://rancher.clusterurl.com/elemental/registration/n99mwjvzjvsdwglcfr75w56pscjltpkpndnzzmx6bz52v76544dqk5
        ca-cert: |
            -----BEGIN CERTIFICATE-----
            MIIBvTCCAWOgAwIBAgIBADAKBggqhkjOPQQDAjBGMRwwGgYDVQQKExNkeW5hbWlj
            bGlzdGVuZXItb3JnMSYwJAYDVQQDDB1keW5hbWljbGlzdGVuZXItY2FAMTY4NDAx
            OTQ4MTAeFw0yMzA1MTMyMzExMjFaFw0zMzA1MTAyMzExMjFaMEYxHDAaBgNVBAoT
            E2R5bmFtaWNsaXN0ZW5lci1vcmcxJjAkBgNVBAMMHWR5bmFtaWNsaXN0ZW5lci1j
            YUAxNjg0MDE5NDgxMFkwEwYHKoZIzj0CAQYIKoZIzj0DAQcDQgAEWg7T8g0xdR6c
            PqBkYe7XIpvic+lf91ZQ4UHSjOx8TGG1UoE2hfM1sXfcz5d4Df3ltnkFxzGzXKfR
            taLaSCuyUqNCMEAwDgYDVR0PAQH/BAQDAgKkMA8GA1UdEwEB/wQFMAMBAf8wHQYD
            VR0OBBYEFP+dYnUy607TQgS6GEOdvwQURyf9MAoGCCqGSM49BAMCA0gAMEUCIDIZ
            4xu9Fy0ZDl4IIwg+2JWLz+nZgpQOKjf8oUyR2TThAiEAuaIlv6B7wbPg1RXO7hol
            bAWW4YDF/W5frbUwC20/TDk=
            -----END CERTIFICATE-----
        emulate-tpm: true
        emulated-tpm-seed: -1
        no-smbios: true
        auth: tpm

```

**Comment by davidcassany** [comment]
@rdconnelly thanks for the detailed report. Could also tell us the elemental-operator version you are using? You can see it in `Installed Apps` tab of the UI on your local cluster and under the `cattle-elemental-system` namespace.

Also, by any chance did you check the associated MachineRegistration resource had a registration URL defined in Status. Looks like the MachineRegistrion was not properly or not yet reconciled and then a seedimage build job was triggered.

**Comment by davidcassany** [comment]
@rdconnelly with the patch from rancher/elemental-operator#517 now it should be impossible to build an ISO without a URL, it would simply error out. The fix will be released as part of v1.4.x elemental operator series by the end of 2023.

One possible explanation to this issue could have been the ISO build was triggered **before** the machine registration was in ready state.

**Comment by fgiudici** [comment]
This should be now fixed, closing the issue
